.github/workflows/ci.yml

```diff
@@ -26,28 +26,63 @@ jobs:
           - "latest-everything"
         test:
           - "default"
-        profile: ["docker"]
+        profile: ["docker", "singularity", "conda"]
+        exclude:
+          - NXF_VER: "latest-everything"
+            profile: "conda"
     steps:
       - name: Check out pipeline code
         uses: actions/checkout@v2
 
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.x"
+
+      - uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+
+      - name: Install Python dependencies
+        run: python -m pip install --upgrade pip pytest-workflow
+
+      - uses: actions/cache@v2
+        with:
+          path: /usr/local/bin/nextflow
+          key: ${{ runner.os }}
+          restore-keys: |
+            ${{ runner.os }}-nextflow-
+
       - name: Install Nextflow
         uses: nf-core/setup-nextflow@v1
         with:
           version: "${{ matrix.NXF_VER }}"
 
-      - name: Set up Python
-        uses: actions/setup-python@v2
+      - name: Set up Singularity
+        if: matrix.profile == 'singularity'
+        uses: eWaterCycle/setup-singularity@v5
         with:
-          python-version: "3.x"
+          singularity-version: 3.7.1
```

Comment:
🤯  why not python anymore? Is this just so it runs with singularity in general?

Reply:
Python is on line 37